substrate/runtime-support/src/storage/mod.rs

```diff
@@ -47,7 +63,7 @@ pub fn get<T: Codec + Sized>(key: &[u8]) -> Option<T> {
 			key: &key[..],
 			pos: 0,
 		};
-		Decode::decode(&mut input).expect("storage is not null, therefore must be a valid type")
+		Decode::decode(&mut AppendZeroes { input: &mut input } ).expect("storage is not null, therefore must be a valid type")
```

Comment:
Virtual dispatch here is really unnecessary and it might have a significant cost if it occurs on every call to `storage::get`